Update to Electron 15

cwillisf opened the pull request:
### Proposed Changes

Update Electron and several support libraries to the latest stable versions:

* electron 8.2.5 -> 15.3.1
   * this is the overall application framework
* electron-builder 22.11.5 -> 22.13.1
   * this controls building and packaging the app installers and store packages
* electron-notarize 0.3.0 -> 1.1.1
   * this is a helper we use to notarize the app for macOS
* electron-devtools-installer 3.0.0 -> 3.2.0
   * this installs dev tools in Electron's browser (only used in debug builds)
* electron-store 5.1.1 -> 8.0.1
   * we use this to store telemetry info, including opt-in/out status

Also, disable `contextIsolation` on browser windows.

### Reason for Changes

Disabling `contextIsolation` causes the browser window to act like it used to, with unrestricted access to Node features like `require`. We should refactor so that we don't need these features, but for now this flag maintains compatibility.

Reasons for package updates:

* Generally keeping up to date
* The old version of `electron-store` didn't support recent versions of `electron`
* The newer version of `electron-builder` has better Windows 11 support
* The newer versions of `electron-notarize` and `electron-builder` have better support for recent versions of macOS

### Test Coverage

I made builds locally and shared them with @BryceLTaylor for manual testing; we haven't found any problems so far.

cwillisf committed:
update electron to 15.3.1

cwillisf committed:
update electron-builder to 22.13.1

cwillisf committed:
update electron-notarize to 1.1.1

cwillisf committed:
update electron-devtools-installer to 3.2.0

cwillisf committed:
update electron-store to 8.0.1

cwillisf committed:
disable contextIsolation for now

We should try to re-enable this once we have time to do the necessary
refactoring

cwillisf commented:
Correction: browser alert windows have a bad title in this version. We'll need to fix that before merging this PR.

The content of the title bar vary. Sometimes it contains remnants of "Scratch" and sometimes it doesn't. Here's an example:

![image](https://user-images.githubusercontent.com/7019101/141007260-a48e90fa-cb01-4774-82c3-dd7ac4aa6e9d.png)

BryceLTaylor commented:
I saw something similar:
![Screenshot (52)](https://user-images.githubusercontent.com/1088291/141007994-9dfa27ed-67ec-48e8-87f4-16d3ab4b5c40.png)

apple502j commented:
That is electron/electron#30223 .

I thought it was some weird mojibake issue, but the comment suggests some issues related to invalid pointer usage. The "fix", as they call it, is to pass `title` explicitly when doing `showMessageBoxSync`.

(also, how many years do we have to wait before we get fully-localized version?)

cwillisf commented:
Thanks for the tip, @apple502j!

(Also, we're definitely planning to localize the rest of the app, but I can't promise a timeline. Sorry it's taking so long!)

cwillisf committed:
work around Electron dialog title bug

In previous versions of this app, Electron automaticaly put "Scratch"
in the title bar for these dialogs.

See https://github.com/electron/electron/issues/30223
